### DISCUSSION TITLE
Code completion stops working in PHPStorm after installing @tailwindcss/ui plugin

### DISCUSSION THREAD

**Discussion by EricBusch** [tailwindlabs/tailwindcss#1854] [Help] [answered]
Hi All

I've installed a default installation of [Jigsaw](https://jigsaw.tighten.co/docs/installation/) and installed tailwindcss via npm. 

At this point, PHPStorm's CSS code completion worked fine.

Then I installed the ui plugin `npm install @tailwindcss/ui` and added `require('@tailwindcss/ui'),` to my tailwind.config.js file.

Now, PHPStorm's code completion stopped working. 

When I comment out the `require()`  statement, PHPStorm's code completion works again.

For some reason, when the `ui` plugin is not installed, PHPStorm seems to use the `build_local` path but when the `ui` plugin is installed, PHPStorm uses the `build_production` path.

I've shared the contents of a few of my files + screenshots of code completion with the `ui` plugin enabled and disabled below.

Any ideas? I suspect it's related to my webpack file that's why I haven't filed a bug... I'm not sure it's not just user error.

### Code completion with plugin disabled

<img width="751" alt="Screen Shot 2020-05-29 at 1 02 08 PM" src="https://user-images.githubusercontent.com/12106436/83285859-01cf5300-a1ad-11ea-82c4-17a31f40b685.png">

### Code completion with plugin enabled

<img width="751" alt="Screen Shot 2020-05-29 at 1 02 57 PM" src="https://user-images.githubusercontent.com/12106436/83285877-085dca80-a1ad-11ea-9871-79d673b65ab5.png">

### webpack.mix.js

```js
const mix = require('laravel-mix');
require('laravel-mix-jigsaw');
const tailwindcss = require('tailwindcss')

mix.disableSuccessNotifications();
mix.setPublicPath('source/assets/build');

mix.jigsaw()
  .js('source/_assets/js/main.js', 'js')
  .options({
    processCssUrls: false,
    postCss: [
      tailwindcss('tailwind.config.js'),
    ]
  })
  .sass('source/_assets/sass/main.scss', 'css')
  .version();
```

### tailwind.config.js

```js
module.exports = {
  purge: [
    './source/**/*.html',
    './source/**/*.vue',
    './source/**/*.jsx',
    './source/**/*.php',
    './source/**/*.blade.php',
  ],
  theme: {
    extend: {},
  },
  variants: {},
  plugins: [
    require('@tailwindcss/ui'),
  ],
}
```

### main.scss

```css
@import "tailwindcss/base";

@import "tailwindcss/components";

@import "tailwindcss/utilities";
```

**Comment by gcw07** [comment, marked as answer]
I had the same thing happen to me last week when I upgraded Tailwind from 1.2, to 1.4. Turns out it was caused by my app.css file compiling together and exceeding the maximum file size PHPStorm would provide code assistance for. The default is like 2 or 3 MB. So if your compiled "main.css" file is larger than that now, it is the cause.

You can change that setting within PHPStorm by going to "Help -> Edit Custom Properties".

Then within that file, add this:

`idea.max.intellisense.filesize=5000`

That is the file size in KB. So set it to whatever is larger than your main.css file size.

**Reply by EricBusch** [reply]
Thank you, Thank you, Thank you! That was it! 👍

**Reply by Filefabrik** [reply]
while working with the tailwind jit
download somewhere the tailwindcss/2.1.2/tailwind.css
and insert into the head-tag the commented link
`<!--<link href="../pattern_tailwind.css" rel="stylesheet" />-->`

**Reply by joseantoniopino** [reply]
This work for me. Thx!
